website/docs/concepts/router.md

```diff
@@ -198,16 +207,161 @@ additional functionality that is not available in `AnyHistory` and
+history changes. Here's how to implement a button that navigates to the `Home` route when clicked.
+
+```rust
+#[function_component(MyComponent)]
+pub fn my_component() -> Html {
+
+    let history = use_history().unwrap();
+
+    let onclick = Callback::once(move |_| history.push(Route::Home));
+
+    html! {
+        <>
+            <button {onclick}>{"Click to go home"}</button>
+        </>
+    }
+}
+```
+:::tip
+The example here uses `Callback::once`. Use a normal callback if the target route can be the same with the route
+the component is in. For example when you have a logo button on every page the that goes back to home when clicked,
+clicking that button twice on home page causes the code to panic because the second click pushes an identical Home route
+and won't trigger a re-render of the element.
+
+In other words, only use `Callback::once` when you are sure the target route is different. Or use normal callbacks only 
+to be safe.
+:::
+
+
+If you want to replace the current history instead of pushing a new history onto the stack, use `history.replace()` 
+instead of `history.push()`.
+
+You may notice `history` has to move into the callback, so it can't be used again for other callbacks. Luckily `history` 
+implements `Clone`, here's for example how to have multiple buttons to different routes:
+
+```rust
+#[function_component(NavItems)]
+pub fn nav_items() -> Html {
+
+    let history = use_history().unwrap();
+
+    let go_home_button = {
+        let onclick = Callback::once({
+            let history = history.clone();
+            move |_| history.push(Route::Home)
+        });
+        html!{<button {onclick}>{"click to go home"}</button>}
+    };
+
+    let go_to_first_post_button = {
+        let onclick = Callback::once({
+            let history = history.clone();
+            let onclick = Callback::once(move |_| history.push(Route::Post{id: "first-post".to_string()}));
+        });
+        
+        html!{<button {onclick}>{"click to go the first post"}</button>}
+    };
+
+    let go_to_secure_button = {
+        let onclick = Callback::once(move |_| history.push(Route::Secure));
+        html!{<button {onclick}>{"click to go to the last post"}</button>}
+    };
+
+    html! {
+        <>
+            {go_home_button}
+            {go_to_first_post_button}
+            {go_to_secure_button}
+        </>
+    }
+}
+```
+
+##### normal components
+
+For normal components, the `AnyHistory` instance can be obtained through the `ctx.link().history()` API. The 
+rest is identical with the functional component case. Here's an example of a view function that renders a single button.
+
+```rust ,ignore
+fn view(&self, ctx: &Context<Self>) -> Html {
+    let history = ctx.link().history().unwrap();
+    let onclick = Callback::once(move |_| history.push(MainRoute::Home));
+    html!{<button {onclick}>{"Go Home"}</button>}
+}
+```
```

Comment:
This doesn't have an issue but is serious. How a normal component can consume the history API was missing from the docs. It felt almost like we are ditching normal components entirely.